doc/api/http.md

```diff
@@ -1520,15 +1537,13 @@ added: v0.3.6
   * `headers` {Object} An object containing request headers.
   * `auth` {String} Basic authentication i.e. `'user:password'` to compute an
     Authorization header.
-  * `agent` {http.Agent|Boolean} Controls [`Agent`][] behavior. When an Agent
-    is used request will default to `Connection: keep-alive`. Possible values:
+  * `agent` {http.Agent|Boolean} Controls [`Agent`][] behavior. Possible values:
    * `undefined` (default): use [`http.globalAgent`][] for this host and port.
    * `Agent` object: explicitly use the passed in `Agent`.
-   * `false`: opts out of connection pooling with an Agent, defaults request to
-     `Connection: close`.
+   * `false`: causes a new `Agent` with default values to be used.
```

Comment:
@mscdex @lpinca not really, no. As I read the code, _all_ `OutgoingMessage`s use `Connection: keep-alive` whether the `keepAlive` option is specified or not. This is what allows the behavior I noted above - that multiple, queued requests will reuse the same socket. So `Connection: keep-alive` is always used, and once the queue of requests has been emptied, then only does `OutgoingMessage` send `Connection: close`.

You can see how sockets are reused even when `keepAlive` option is false [here](https://github.com/nodejs/node/blob/master/lib/_http_agent.js#L47-L54).

As I've said before, I could be mistaken and reading the code wrong, but nobody has actually given me another interpretation of the code itself that corresponds to the old documentation. My text changes are based on what I see in the code. I've tried to provide that analysis [here](https://github.com/nodejs/node/pull/10715#issuecomment-272241686) and [here](https://github.com/nodejs/node/issues/10590). If you think it's wrong or mistaken, please let me know how and where. The code is spaghettiish and quite confusing.